sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/CosmosTemplate.java

```diff
@@ -58,34 +59,42 @@
     private static final Logger LOGGER = LoggerFactory.getLogger(CosmosTemplate.class);
 
     private final MappingCosmosConverter mappingCosmosConverter;
-    private final String databaseName;
-    private final ResponseDiagnosticsProcessor responseDiagnosticsProcessor;
-    private final boolean enableQueryMetrics;
```

Comment:
We can keep these three instance variables and populate them in the constructor instead of getting them from CosmosFactory and CosmosConfig - that way, throughout CosmosTemplate we can just use these instance variables and not get them every time using the APIs.